### DISCUSSION TITLE
Automated OpenAPI Documentation

### DISCUSSION THREAD

**Discussion by petecheslock** [mastodon/mastodon#22314] [General]
There is an existing Issue opened for this already (https://github.com/mastodon/mastodon/issues/20572) - but figured posting here could be useful for other people who were interested in getting access to the Mastodon OpenAPI specification. 

I wrote a post recent showing how we can use [AppMap for Ruby](https://github.com/getappmap/appmap-ruby) to record the Mastodon rspec test cases and then export this data to an OpenAPI v3 specification file.  

Let me know if this is helpful to anyone.  https://dev.to/appmap/automatically-generating-openapi-docs-for-mastodon-42ig

**Comment by abraham** [comment]
If anyone's interested I've got a spec generated by parsing the Mastodon docs. https://abraham.github.io/mastodon-openapi/
